client/allocrunner/taskrunner/task_runner.go

```diff
@@ -490,6 +593,11 @@ func (tr *TaskRunner) persistLocalState() error {
 	return tr.stateDB.PutTaskRunnerLocalState(tr.allocID, tr.taskName, tr.localState)
 }
 
+// buildID builds a consistent unique ID for the task from the alloc ID, task name and restart attempt
+func (tr *TaskRunner) buildID() string {
+	return fmt.Sprintf("%s/%s/%d", tr.allocID, tr.taskName, tr.restartTracker.GetCount())
```

Comment:
I don't think so. When the hooks run it should destroy the existing task so that ID can then be reused. This is mainly to avoid a situation where the task has exited and the client restarts it (by creating a new process) before cleanup of the old is finished. I'm not sure if this scenario can happen or not.